app/src/sharedTest/java/org/oppia/app/utility/DragViewAction.kt

```diff
@@ -0,0 +1,204 @@
+package org.oppia.app.utility
+
+import android.os.SystemClock
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewConfiguration
+import androidx.annotation.Nullable
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.PerformException
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.CoordinatesProvider
+import androidx.test.espresso.action.MotionEvents
+import androidx.test.espresso.action.PrecisionDescriber
+import androidx.test.espresso.action.Swiper
+import androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed
+import androidx.test.espresso.util.HumanReadables
+import org.hamcrest.Matcher
+import org.hamcrest.core.AllOf.allOf
+
+/**
+ * The number of move events to send for each drag.
+ */
+const val DRAG_STEP_COUNT = 10
+/**
+ * Length of time a drag should last for, in milliseconds.
+ */
+private const val DRAG_DURATION = 1500
+
+/**
+ * Duration between the last move event and the up event, in milliseconds.
+ */
+private const val WAIT_BEFORE_SENDING_UP = 400
+
+class DragViewAction(
+  private val mDragger: Dragger,
+  private val mStartCoordinatesProvider:CoordinatesProvider,
+  private val mEndCoordinatesProvider: CoordinatesProvider,
+  private val mPrecisionDescriber: PrecisionDescriber
+) : ViewAction {
+
+  interface Dragger : Swiper {
+    fun wrapUiController(uiController: UiController): UiController
+  }
+
+  /**
+   * Executes different drag types to given positions.
+   */
+  enum class Drag : Dragger {
+    /**
+     * Starts a drag with a long-press.
+     */
+    LONG_PRESS {
+      private val downMotion: DownMotionPerformer =
+        object : DownMotionPerformer {
+          override fun perform(uiController: UiController, coordinates: FloatArray, precision: FloatArray): MotionEvent {
+            return performLongPress(uiController,coordinates,precision)
```

Comment:
Spacing here as well, and elsewhere in this class.

Reply:
done